docs/source/configuration/persisted-queries.mdx

```diff
@@ -66,6 +66,18 @@ persisted_queries:
 
 If used with the [`safelist`](#safelist) option, the router logs unregistered and rejected operations. With [`safelist.required_id`](#require_id) off, the only rejected operations are unregistered ones. If [`safelist.required_id`](#require_id) is turned on, operations can be rejected even when registered because they use operation IDs rather than operation strings.
 
+#### `experimental_local_manifest`
+
+<ExperimentalFeature />
+
+Adding `experimental_local_manifest` to your `safelist` configuration allows you to use a local persisted query manifest in place of the hosted Uplink version. This is useful when you have an offline Enterprise license and are unable to use Uplink. This configuration option cause the router to not reload the manifest from the file system, so you will need to restart the router to apply changes.
```

Comment:
Should we link to an example of what such a file looks like?

Reply:
We could, I suppose I would be curious on how we do that/if there's a better resource that we could link to instead for this. @apollographql/docs - thoughts?

Reply:
Hmm, AFAIK we don't have a standard protocol for this. I could see us linking to a repo of maintained examples/solutions if something like that exists.

In this case, though the example isn't particularly fleshed out, what about linking to the `apollo-router/tests/fixtures/persisted-queries-manifest.json` that's part of this PR just so readers can see the expected format?

Reply:
That may be sane- I suppose my only comment is that they shouldn't be crafting it at all, but rather using this here: 
![image](https://github.com/apollographql/router/assets/5595530/8c9c7a56-e476-4c46-9949-624d774a379b)

Perhaps it's just using a screenshot like that?

Reply:
> That may be sane- I suppose my only comment is that they shouldn't be crafting it at all, but rather using this here:

Oh, that's very helpful information to callout! I think that deserves it's own explanation. Will add a suggestion for that below the current example config momentarily.
```

Reply:
One note about that UI control. On the GraphOS -> Router side, we divide a PQL up into "chunks" (each having the format of a manifest JSON) that Router can download separately to limit the size of each individual download (and allow it to only refresh the part of the list that changed).  (Chunk division is not related to the order or structure of how the operations were uploaded.) For better or for worse I believe that the download button in the UI tries to download all of the chunks (in one GraphQL operation, so hopefully that scales well enough) and then concatenates them (without a separator, I think) into the file.

In the common case this is just one JSON object but as folks scale it will eventually become two JSON objects in the same file and so this will fail.

Thoughts on what we should do here? We could change the UI to do something slightly different like parse and join the files or at least insert newlines between them (though at the end of the day I'm not sure how great an idea it is to try to offer in the website a single-click one-file download of something we've explicitly architected as a sequence of bounded-size files).  Or you could change your code to understand that the file might be a "more than one concatenated JSON object" file.  But if we recommend this button and don't make another change, this will start failing for people eventually (either your code won't be able to parse the file, or it'll just ignore part of the file after the first object or something.

Reply:
That's... interesting. I'd prefer we just concat client-side and send a larger JSON, but if the format is changing- that's another concern altogether since there's not a good way to handle this as far as I know in Rust beyond checking for one format, then another for parsing. 

Anyways, happy to do whatever, just probably should figure out the right path before we dig our own grave.

Reply:
From our conversation yesterday, I've pushed up a change to make this take an array of manifests vs. just one- should allow us to account for the chunking situation as well as single-file cases.